### PR TITLE
KAFKA-16076: Fix Missing thread interrupt call in RestClient class 

### PR DESCRIPTION
**Pull request by abhi-ksolves**
In RestClient class, httpRequest is being called with different threads. An InterruptedException is used to handle specific exceptions in case of failure. It's a good practice to call Thread.currentThread().interrupt() so the rest of the code knows the thread was interrupted. A separate catch block is added to handle InterruptedException & interrupted the current thread.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ HttpClient httpClient(SslContextFactory.Client sslContextFactory) {
      * @return The deserialized response to the HTTP request, containing null if no data is expected or returned.
      */
     public <T> HttpResponse<T> httpRequest(String url, String method, HttpHeaders headers, Object requestBodyData,
-                                                  TypeReference<T> responseFormat) {
```
Comment (chia7712) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java]: Could you please revert those unrelated changes? It makes this PR smaller :)

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java [hunk 2]
```diff
@@ -189,9 +190,13 @@ private <T> HttpResponse<T> httpRequest(HttpClient client, String url, String me
                         Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
                         "Unexpected status code when handling forwarded request: " + responseCode);
             }
-        } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
+        } catch (IOException | TimeoutException | ExecutionException e) {
             log.error("IO error forwarding REST request to {} :", url, e);
             throw new ConnectRestException(Response.Status.INTERNAL_SERVER_ERROR, "IO Error trying to forward REST request: " + e.getMessage(), e);
+        } catch (InterruptedException e) {
+            log.error("Thread was interrupted forwarding REST request to {} :", url, e);
+            Thread.currentThread().interrupt();
```
Comment (chia7712) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java:197]: Could you please give a try of adding unit test?

**Review by chia7712 — COMMENTED** [review]
@abhi-ksolves thanks for this patch. 

BTW, please revise the title. It should start with either `MINOR:` or jira number.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java (hunk 2 above)
Comment (abhi-ksolves) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java:197]: I believe it is not required because this httpRequest is an overloaded method that is only used by another httpRequest method invoked in RestClientTest.java.

**Comment by gharris1727** [comment]
Supersedes #15110

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java (hunk 2 above)
Comment (chia7712) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java:197]: > I believe it is not required because this httpRequest is an overloaded method that is only used by another httpRequest method invoked in RestClientTest.java.

not sure whether I catch your point. It seems the `httpRequest` you fix is used by other variety `httpRequest` of `RestClient`. It is used by production, and so it is worthwhile to have a simple test case in `RestClientTest`. For example:
```java
        @Test
        public void testInterruptCall() throws ExecutionException, InterruptedException, TimeoutException {
            Request req = mock(Request.class);
            doThrow(new InterruptedException()).when(req).send();
            doReturn(req).when(req).header(anyString(), anyString());
            doReturn(req).when(httpClient).newRequest(anyString());
            ConnectRestException e = assertThrows(ConnectRestException.class, () -> httpRequest(
                httpClient, MOCK_URL, TEST_METHOD, TEST_TYPE, TEST_SIGNATURE_ALGORITHM
            ));
            assertIsInternalServerError(e);
            assertInstanceOf(InterruptedException.class, e.getCause());
            assertTrue(Thread.interrupted());
        }
```

Comment (abhi-ksolves) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java:197]: this test case seems suitable i have added the same

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
all failed tests pass on my local. will merge it
